config/main.py

```diff
@@ -1893,43 +1892,35 @@ def portchannel(db, ctx, namespace):
 @click.pass_context
 def add_portchannel(ctx, portchannel_name, min_links, fallback):
     """Add port channel"""
-    if is_portchannel_name_valid(portchannel_name) != True:
-        ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}'"
-                 .format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO))
-
+    
     db = ctx.obj['db']
-
+    validated_db = ValidatedConfigDBConnector(db)
+    
     if is_portchannel_present_in_db(db, portchannel_name):
         ctx.fail("{} already exists!".format(portchannel_name))
-
+    
     fvs = {'admin_status': 'up',
-           'mtu': '9100',
+           'mtu': '9100',  
            'lacp_key': 'auto'}
     if min_links != 0:
         fvs['min_links'] = str(min_links)
     if fallback != 'false':
         fvs['fallback'] = 'true'
-    db.set_entry('PORTCHANNEL', portchannel_name, fvs)
-
+    validated_db.set_entry('PORTCHANNEL', portchannel_name, fvs)
+    
 @portchannel.command('del')
 @click.argument('portchannel_name', metavar='<portchannel_name>', required=True)
 @click.pass_context
 def remove_portchannel(ctx, portchannel_name):
     """Remove port channel"""
-    if is_portchannel_name_valid(portchannel_name) != True:
-        ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}'"
-                 .format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO))
 
     db = ctx.obj['db']
-
-    # Dont proceed if the port channel does not exist
-    if is_portchannel_present_in_db(db, portchannel_name) is False:
-        ctx.fail("{} is not present.".format(portchannel_name))
+    validated_db = ValidatedConfigDBConnector(db)
 
     if len([(k, v) for k, v in db.get_table('PORTCHANNEL_MEMBER') if k == portchannel_name]) != 0:
         click.echo("Error: Portchannel {} contains members. Remove members before deleting Portchannel!".format(portchannel_name))
```

Comment:
>Portchannel {} contains members [](http://example.com/codeflow?start=27&length=31)

Can we depend yang model to validate this? If capable, you can remove this if branch.

Reply:
The check exists, though the error message is not intuitive: "Data Loading failed, points to a non-existing leaf." 
I can remove this if branch for now and clarify the error messages at a later point